### PR TITLE
Make Dispatcher per Context from per Thread, Simplify TestHarnessRunner

### PR DESCRIPTION
**Pull request by skmp**
#### Overview

Cleanup/refactor. There were very few things left per Thread, so this was fairly painless.

TestHarnessRunner used the `CPUFactory` interface and a custom core to run the tests on the host, however the context and all related classes aren't needed or used for this. It has been reworked to directly run the host tests without a FEX Context.

#### Motivation
Apart from a nice cleanup, this is yet another incremental step towards context-sharable codegen, which can work nicely in combination with object code caching.

I have a PoC of context-sharable codegen in `skmp/context-wide-buffering` though it is buggy and leaky.

#### Depends on
#1770

#### Splits
#1778

### CODE REVIEW

**Comment by skmp** [comment]
(Rebased for #1775 split)

File: External/FEXCore/Source/Interface/Core/Core.cpp [hunk 1]
```diff
@@ -703,151 +712,160 @@ namespace FEXCore::Context {
...
-    Thread->PassManager->Run(Thread->OpDispatcher.get());
+    Thread->PassManager->Run(IREmitter);
 
     // Debug
     {
       if (Thread->CTX->Config.DumpIR() != "no") {
-        IRDumper(Thread, GuestRIP, Thread->PassManager->HasPass("RA") ? Thread->PassManager->GetPass<IR::RegisterAllocationPass>("RA")->GetAllocationData() : nullptr);
-      }
-
-      if (Thread->OpDispatcher->ShouldDump) {
```
Comment (Sonicadvance1) [inline External/FEXCore/Source/Interface/Core/Core.cpp:838]: What replaces this `ShouldDump` usage?
This is highly useful for new instruction implementations to have instructions that are new emit their blocks of IR implementation.

**Review by Sonicadvance1 — APPROVED** [review]
Apart from the single comment this looks fine.

File: External/FEXCore/Source/Interface/Core/Core.cpp (hunk 1 above)
Comment (skmp) [reply External/FEXCore/Source/Interface/Core/Core.cpp:838]: And I /thought/ it might be so. Will address.

**Pull request merged**